app/services/catalog/order_item_sanitized_parameters.rb

```diff
@@ -42,26 +42,26 @@ def sanitized_parameters
       svc_params = ActiveSupport::HashWithIndifferentAccess.new(service_parameters)
       if service_plan_schema[:properties].present?
         service_plan_schema[:properties].each_with_object({}) do |(key, attrs), result|
-          value = hide?(key, attrs) ? MASKED_VALUE : svc_params[key]
+          value = mask_ocp_value?(key, attrs) ? MASKED_VALUE : svc_params[key]
           result[attrs[:title]] = value
         end
       else
-        service_plan_schema[:schema][:fields].each_with_object({}) do |(field), result|
-          value = hide_ansible?(field) ? MASKED_VALUE : svc_params[field[:name]]
+        service_plan_schema[:schema][:fields].each_with_object({}) do |field, result|
+          value = mask_ansible_value?(field) ? MASKED_VALUE : svc_params[field[:name]]
           result[field[:name]] = value
         end
       end
     end
 
-    def hide_ansible?(field)
-      FILTERED_PARAMS.reduce(field[:type] == "password") do |result, param|
-        result || /#{param}/i.match?(field[:name]) || /#{param}/i.match?(field[:label])
+    def mask_ocp_value?(key, attrs)
+      FILTERED_PARAMS.reduce(attrs[:format] == "password") do |result, param|
+        result || /#{param}/i.match?(attrs[:title]) || /#{param}/i.match(key)
       end
     end
 
-    def hide?(key, attrs)
-      FILTERED_PARAMS.reduce(attrs[:format] == "password") do |result, param|
-        result || /#{param}/i.match?(attrs[:title]) || /#{param}/i.match(key)
+    def mask_ansible_value?(field)
```

Comment:
These `mask_*_value` methods would be more accurately named `mask_json_schema_value?` and `mask_ddf_schema_value?`.  The format is not tied to the source type, it is the schema type.

Reply:
After chatting with @lindgrenj6 I think we might be missing an important detail.  First off, `DDF` refers to "Data-Driven-Form) which is the UI component rendering the dialog.  This component supports different formats, include JSON schema as well as it's own schema designed to be more UI-aware.

Second, Topology is designed to be a normalization layer for other micro-services.  The dialog format coming back should be a consistent type regardless of the source.  The reason we are dealing with this issue today is that we have not completely remove the deprecated OCPv3 service plan which used the JSON schema.

Going forward, all new source types would represent dialogs in the DDF schema.  There really is no need to support the OCP format anymore.  Yes, OCP items will fail to create an approval request, but the call is never going to succeed anyway since we remove the namespace selection.  The OCP logic will eventually be replaced by a newer OCP implementation (likely Operator based) that would utilize the DDF schema.

The end result is maybe we rename `mask_ansible_value?` to just `mask_value?` and drop the `ocp` checks.
Note: There was talk about converting the existing OCPv3 dialogs to the new format, but it is more likely we will remove the Topology collection of OCPv3 service_offering items completely so it is not worth the effort at the moment.

cc @syncrou @eclarizio

Reply:
Implemented this.